.github/workflows/atfe_pre_merge.yml

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2025, Arm Limited and affiliates.
+# Part of the Arm Toolchain project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# This workflow runs automatically when a PR is opened, updated, or reopened against:
+#   - arm-toolchain branch
+#   - any release/arm-software/** branch
+#
+# It builds and tests the Arm Toolchain for Embedded (ATfE) to ensure that the
+# changes do not introduce regressions and that the build remains stable before
+# merging.
+
+name: ATfE Pre-Merge
+
+on:
+  # Trigger when a pull request is opened, updated, or reopened
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - arm-toolchain
+      - release/arm-software/**
+
+jobs:
+  build-and-test:
```

Comment:
I think we'll need to add a protection layer to this workflow to require maintainer approval before executing the workflow.
From what I checked, this can be done using GitHub's Environemnts:
- https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-deployments/managing-environments-for-deployment
- https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idenvironment